contracts/Colony.sol

```diff
@@ -54,26 +54,19 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
     return ColonyAuthority(address(authority)).hasUserRole(_user, _role);
   }
 
-  function getColonyNetworkAddress() public view returns (address) {
+  function getColonyNetwork() public view returns (address) {
     return colonyNetworkAddress;
   }
 
-  function setToken(address _token) public
-  stoppable
-  auth
-  {
-    token = ERC20Extended(_token);
-
-    emit ColonyTokenSet(_token);
-  }
-
   function getToken() public view returns (address) {
     return address(token);
   }
 
-  function initialiseColony(address _colonyNetworkAddress) public stoppable {
-    require(colonyNetworkAddress == address(0x0), "colony-initialise-bad-address");
+  function initialiseColony(address _colonyNetworkAddress, address _token) public stoppable {
+    require(colonyNetworkAddress == address(0x0), "colony-already-initialised-network");
+    require(address(token) == address(0x0), "colony-already-initialised-token");
```

Comment:
Q: why cast `address(token)` if `token` is already an `address` type?

Reply:
A: But it's not an address type: https://github.com/JoinColony/colonyNetwork/blob/develop/contracts/ColonyStorage.sol#L35

Reply:
Hmm. Mixing address and contract types for variables seems like a recipe for confusion. Is there a reason we don't store everything as `address` types and cast them to their specific contract types when we need to use them? Some sort of convention around this seems appropriate.